### ISSUE TITLE
In BaseTester, merge the "fit" and "transform" calls to "fit_transform"

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#60]
```fit_transform``` seems more universal, as sklearn.manifold.TSNE has ```fit_transform``` and ```fit``` but not ```transform```

**Comment by KevinMusgrave** [comment]
Available in v0.9.85.dev0
